src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-details/osd-details.component.spec.ts

```diff
@@ -23,19 +30,50 @@ describe('OsdDetailsComponent', () => {
       DataTableModule,
       SharedModule
     ],
-    declarations: [OsdDetailsComponent, OsdPerformanceHistogramComponent]
+    declarations: [OsdDetailsComponent, OsdPerformanceHistogramComponent],
+    providers: [OsdService]
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(OsdDetailsComponent);
     component = fixture.componentInstance;
 
     component.selection = new CdTableSelection();
+    debugElement = fixture.debugElement;
+    osdService = debugElement.injector.get(OsdService);
+
+    getDetailsSpy = spyOn(osdService, 'getDetails');
 
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should fail creating a histogram', () => {
+    const detailDataWithoutHistogram = {
+      osd_map: {},
+      osd_metadata: {},
+      histogram: 'osd down'
+    };
+    getDetailsSpy.and.returnValue(of(detailDataWithoutHistogram));
+    component.osd = { tree: { id: 0 } };
+    component.refresh();
+    expect(getDetailsSpy).toHaveBeenCalled();
+    expect(component.osd.histogram_failed).toBe('osd down');
+  });
+
+  it('should succeed creating a histogram', () => {
+    const detailDataWithHistogram = {
+      osd_map: {},
+      osd_metdata: {},
+      histogram: {}
+    };
+    getDetailsSpy.and.returnValue(of(detailDataWithHistogram));
+    component.osd = { tree: { id: 0 } };
+    component.refresh();
+    expect(getDetailsSpy).toHaveBeenCalled();
+    expect(component.osd.histogram_failed).toBe('');
```

Comment:
If you expect it to be `undefined` here, you can remove line 37 in `osd-details.component.ts`

Reply:
Yes, I know. And although possible, I prefer to properly initialize variables when using statically typed languages. Maybe it's just a matter of preference, but to me it's clearer this way.

Do you have a strong opinion about that and wan't me to change it?

Reply:
An empty string would be considered true standing alone in a condition in TS. As the attribute is not used if the histogram is there, I suggest to not to initialize it at the point where you don't know if it is needed.